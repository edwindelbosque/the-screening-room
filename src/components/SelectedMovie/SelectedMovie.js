import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "./SelectedMovie.scss";
import { TrailerPlayer } from '../TrailerPlayer/TrailerPlayer';
import { getTrailer } from '../../apiCalls/apiCalls';

class SelectedMovie extends Component {
  constructor() {
    super()
    this.state = {
      videoKey: ''
    }
  }
  componentDidMount = async () => {
    this.setState({
      videoKey: await getTrailer(this.props.movieDetails.movie_id) ? await getTrailer(this.props.movieDetails.movie_id) : ''
    })
  }

  render() {
    const { wallpapers, movieDetails, match } = this.props;
    const { title, overview, poster_path, rating, movie_id, release_date } = movieDetails;
    const movieWallpaper = wallpapers.find(wallpaper => wallpaper.id === movie_id)
      .wallpaper;
    return (
      <>
        <Link to={`/${match.params[0]}`}>
          <div className="modal-backdrop-white">
            <div className="modal-backdrop"></div>
          </div>
        </Link>
        <main className="SelectedMovie movie-modal">
          <div className="SelectedMovie__div--container">
            <header className="SelectedMovie__header">
              {
                this.state.videoKey ? <TrailerPlayer youtubeId={this.state.videoKey} className="SelectedMovie__img--favorite" />
                  : <img
                    className="SelectedMovie__img--favorite"
                    src={movieWallpaper}
                    alt="Wallpaper"
                  />
              }
              <img
                className="SelectedMovie__img--close"
                src={poster_path}
                alt="Close window icon"
              />
            </header>
            <div className="bottom-container">
              <div className="movie-text">
                <h2><span>{title}</span></h2>
                <h3 className="SelectedMovie__h3">{overview}</h3>
              </div>
              <div className="movie-details">
                <h4>
                  Rating <span>{rating}</span>
                </h4>
                <h4 className='release-date'>
                  Released <span>{release_date.slice(0, 4)}</span>
                </h4>
              </div>
            </div>
          </div>
        </main>
      </>
    );
  };
}

export default SelectedMovie;

SelectedMovie.propTypes = {
  movieDetails: PropTypes.object,
  wallpapers: PropTypes.array
  // match: Proptypes.object
};
