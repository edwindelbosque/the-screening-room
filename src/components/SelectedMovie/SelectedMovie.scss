.SelectedMovie__img--favorite {
  width: 100%;
  position: relative;
  border-radius: 4px 4px 0 0;
  object-fit: cover;
  height: 400px;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.514);
}

.SelectedMovie__img--close {
  width: 200px;
  position: absolute;
  bottom: 110px;
  left: 32px;
  border-radius: 3px;
  border: 1px solid rgba(58, 58, 58, 0.863);
  border-bottom: none;
  box-shadow: 0 2px 3px rgba(0, 0, 0, 0.801);
}

.movie-modal {
  position: fixed;
  z-index: 100;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
  display: flex;
  flex-direction: column;
  background-color: white;
  border-radius: 5px 5px 4px 4px;
  border: none;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.479);
  min-width: 20rem;
  max-width: 60rem;
  width: 800px;
  height: 675px;
  background: rgb(255, 255, 255);
  border-bottom: 8px solid #01C369;

  h2 {
    color: rgb(255, 255, 255);
    font-size: 29px;
    z-index: 1;
  }

  h3 {
    margin-top: 30px;
    font-size: 17px;
    color: rgb(48, 48, 48);
    font-weight: 500;
  }

  .movie-details {
    display: flex;
    flex-direction: column;
    margin-left: 32px;
    position: relative;
    top: 165px;
    font-weight: 300;
  }

  h4 {
    color: rgb(63, 63, 63);
    font-weight: 500;
    padding-bottom: 5px;
    width: 200px;

    span {
      font-weight: 300;
    }
  }

}

.movie-text {
  width: 70%;
  position: relative;
  margin-right: 50px;
  left: 20px;
  bottom: 50px;
}

.bottom-container {
  display: flex;
  justify-content: space-between;
  flex-direction: row-reverse;
}