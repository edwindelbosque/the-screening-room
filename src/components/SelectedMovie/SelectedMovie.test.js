import React from 'react';
import { shallow } from 'enzyme';
import SelectedMovie from './SelectedMovie';

describe('SelectedMovie', () => {
  const wallpapers = [
    { id: 1, wallpaper: 'path1' },
    { id: 2, wallpaper: 'path2' }
  ];
  const match = { params: ['location1', 'location2'] };
  let wrapper;
  let movieDetails = {
    movie_id: 2,
    wallpaper:
      'https://image.tmdb.org/t/p/original/skvI4rYFrKXS73BJxWGH54Omlvv.jpg',
    poster:
      'https://image.tmdb.org/t/p/original/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg',
    title: 'Maleficent: Mistress of Evil',
    overview:
      'Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.',
    genre: ['Adventure', 'Fantasy', 'Family'],
    rating: 7.2,
    release_date: '2019-06-09'
  };

  beforeEach(() => {
    wrapper = shallow(
      <SelectedMovie
        movieDetails={movieDetails}
        wallpapers={wallpapers}
        match={match}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
