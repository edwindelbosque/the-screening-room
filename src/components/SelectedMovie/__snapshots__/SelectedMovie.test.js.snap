// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMovie should match the snapshot 1`] = `
<Fragment>
  <Link
    to="/location1"
  >
    <div
      className="modal-backdrop-white"
    >
      <div
        className="modal-backdrop"
      />
    </div>
  </Link>
  <main
    className="SelectedMovie movie-modal"
  >
    <div
      className="SelectedMovie__div--container"
    >
      <header
        className="SelectedMovie__header"
      >
        <img
          alt="Wallpaper"
          className="SelectedMovie__img--favorite"
          src="path2"
        />
        <img
          alt="Close window icon"
          className="SelectedMovie__img--close"
        />
      </header>
      <div
        className="bottom-container"
      >
        <div
          className="movie-text"
        >
          <h2>
            <span>
              Maleficent: Mistress of Evil
            </span>
          </h2>
          <h3
            className="SelectedMovie__h3"
          >
            Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.
          </h3>
        </div>
        <div
          className="movie-details"
        >
          <h4>
            Rating 
            <span>
              7.2
            </span>
          </h4>
          <h5
            className="release-date"
          >
            <span>
              2019
            </span>
          </h5>
        </div>
      </div>
    </div>
  </main>
</Fragment>
`;
